### PR TITLE
Make MLeap Spark 3.0 build work with XGBoost 1.0.0

### PR DESCRIPTION
**Pull request by WeichenXu123**
This is bug in xgboost-predictor library, they have been fixed in these PRs. 
https://github.com/h2oai/xgboost-predictor/pull/14
https://github.com/h2oai/xgboost-predictor/pull/15

Now I upgrade xgboost-predictor library and xgboost test passed.

### CODE REVIEW

File: mleap-spark-testkit/src/main/scala/org/apache/spark/ml/parity/SparkParityBase.scala [hunk 1]
```diff
@@ -166,7 +166,7 @@ abstract class SparkParityBase extends FunSpec with BeforeAndAfterAll {
     }
   }
 
-  var relTolEps: Double = 1E-30
+  var relTolEps: Double = 1E-7
```
Comment (WeichenXu123) [inline mleap-spark-testkit/src/main/scala/org/apache/spark/ml/parity/SparkParityBase.scala:169]: The change 1E-30 is merged from master code. It break Scalar transformer test.
@ancasarb 
1E-30 is too small do not fit our numerical error. We should also update it in master.

Comment (mengxr) [reply mleap-spark-testkit/src/main/scala/org/apache/spark/ml/parity/SparkParityBase.scala:169]: Could you send a PR to master?

File: project/Dependencies.scala [hunk 2]
```diff
@@ -72,7 +72,7 @@ object Dependencies {
     val kryo = "com.esotericsoftware" % "kryo" % kryoVersion
     // exclude jar "com.esotericsoftware.kryo % kryo" which conflicts with spark 3.0
     val xgboostDep = "ml.dmlc" %% "xgboost4j" % xgboostVersion exclude("com.esotericsoftware.kryo", "kryo")
-    val xgboostPredictorDep = "biz.k11i" % "xgboost-predictor" % "0.3.1" exclude("com.esotericsoftware.kryo", "kryo")
+    val xgboostPredictorDep = "ai.h2o" % "xgboost-predictor" % "0.3.17" exclude("com.esotericsoftware.kryo", "kryo")
```
Comment (mengxr) [inline project/Dependencies.scala:75]: I saw the last update from https://github.com/komiya-atsushi/xgboost-predictor-java was Nov 26, 2019. Does H2O's fork contain changes that won't go back to upstream?

cc: @komiya-atsushi @honzasterba

File: mleap-spark-testkit/src/main/scala/org/apache/spark/ml/parity/SparkParityBase.scala (hunk 1 above)
Comment (WeichenXu123) [reply mleap-spark-testkit/src/main/scala/org/apache/spark/ml/parity/SparkParityBase.scala:169]: Create PR in master.
https://github.com/combust/mleap/pull/687

File: project/Dependencies.scala (hunk 2 above)
Comment (honzasterba) [reply project/Dependencies.scala:75]: Hi, I did not plan to bring our changes to the original repo. I think the two diverged a little even before I started working on H2O.

Comment (WeichenXu123) [reply project/Dependencies.scala:75]: @komiya-atsushi Do you want to support xgboost 1.0 in your repo ?

Comment (mengxr) [reply project/Dependencies.scala:75]: I merged the change into spark-3.0.0 branch. We might want to switch back to the official release once it supports xgboost 1.0.0.

**Pull request merged**